plugin-api/src/main/java/org/hyperledger/besu/plugin/BesuPlugin.java

```diff
@@ -52,4 +64,13 @@
    * started.
    */
   void stop();
+
+  /**
+   * Called when the plugin is being reloaded. This method will be called trough a dedicated JSON
+   * RPC endpoint. If not overridden this method does nothing for convenience. The plugin should
+   * only implement this method if it supports dynamic reloading.
+   *
+   * <p>The plugin should reload its configuration dynamically or do nothing if not applicable.
+   */
+  default void reloadConfiguration() {}
```

Comment:
I named it reload first. But after i realized that it may be confusing for the user because he may think that it reloads the whole plugin including the binary if it has been changed after Besu startup. I am happy to change to reload again if you think it is better.